src/v2/client/interact.js

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) 2020, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+// BaseLoginRouter contains the more complicated router logic - rendering/
+// transition, etc. Most router changes should happen in LoginRouter (which is
+// responsible for adding new routes)
+
+import idx from 'idx';
+
+import { getTransactionMeta, saveTransactionMeta } from './transactionMeta';
+
+// Begin or resume a transaction using an interactionHandle
+export async function interact (settings) {
+  const clientId = settings.get('clientId');
+  const domain = settings.get('baseUrl');
+  const scopes = settings.get('scopes');
```

Comment:
Are we addressing this `scopes` vs `authParams.scopes` in a different PR?

Reply:
That is a separate issue (OKTA-361428), that affects `showSignInToGetTokens`.

There may be a similar issue here. I will see if some more unit tests are needed here.

Reply:
I am now pulling all theses values from the transaction meta which is set by the authClient. This should avoid any possible inconsistency due to weird widget config.